### PR TITLE
[add] Form View

### PR DESCRIPTION
**Pull request by edbella**
Created the basic form view showing email and password fields, connected to redux store and firebase auth.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import React, { Component } from 'react';
 import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
 import AOS from 'aos';
+import { Provider } from 'react-redux';
 import Homepage from './components/homepage/components';
 import Navbar from './components/navbar/components';
```
Comment (justiceotuya) [inline src/App.jsx]: Do all exports from the hompage file and navbar file. Eg
In homepage/components have an index.js file that imports individual componets and exports these component. Then in the homepage folder, havr another index.js that imports everything from the components folder and exports everything. So any file that wants to reference the file from the hompage components will just do import example from ./components/homepage

File: src/App.jsx [hunk 2]
```diff
@@ -33,14 +35,16 @@ class App extends Component {
...
-                    <Route path="/" exact component={Homepage} />
-                    <Route path="/sign-up" exact component={Signup} />
-                    <Route path="/sign-in" exact component={SignIn} />
-                </Switch>
-            </Router>
+            <Provider store={store}>
+                <Router>
+                    <Navbar />
+                    <Switch>
+                        <Route path="/" exact component={Homepage} />
```
Comment (justiceotuya) [inline src/App.jsx]: This is where @enyeInc will say dont repeat yourself

Comment (edbella) [reply src/App.jsx]: We already have plans to fix that. Trying to do that now might exceed the 250+ limit so its been planned for the next PR.

Comment (edbella) [reply src/App.jsx]: @justiceotuya

File: src/components/signin/actionTypes.js [hunk 3]
```diff
@@ -8,3 +8,5 @@ export const LOGOUT_FAILURE = 'LOGOUT_FAILURE';
 
 export const VERIFY_REQUEST = 'VERIFY_REQUEST';
 export const VERIFY_SUCCESS = 'VERIFY_SUCCESS';
```
Comment (justiceotuya) [inline src/components/signin/actionTypes.js]: try this syntax

export const actionTypes = {
VERIFY_REQUEST: 'VERIFY_REQUEST', 
VERIFY_SUCCESS: 'VERIFY_SUCCESS', 
}
Then you can then import it like 'import actionTypes from actionTypes' 

Then destruct and use


Sorry i am reviewing from my phone

File: src/components/signin/actions.js [hunk 4]
```diff
@@ -25,9 +25,9 @@ export const requestLogin = () => ({ type: LOGIN_REQUEST });
...
  * @param {Object} User - The user object received from firebase on login.
  */
-export const receiveLogin = async user => {
+export const receiveLogin = async payload => {
     const fbfunction = process.env.REACT_APP_USER_AUTH_API;
-    const userInfo = { userAuth: user, userType: 'user' };
+    const userInfo = { userAuth: payload, userType: 'user' };
     const config = { headers: { 'Content-Type': 'application/json' } };
     const body = JSON.stringify(userInfo);
     const res = await axios.post(fbfunction, body, config);
```
Comment (justiceotuya) [inline src/components/signin/actions.js]: You are suppose to make all side effect request with sagas. All calls to an external source should be via sagas

File: src/components/signin/components/AntdForm.jsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    const { signIn: { email, password } } = data;
+    const user = { email, password };
+
+    /**
+    * @function
+    * @param {event} e - Listening for an event
+    */
+    function handleSubmit(e) {
+        e.preventDefault();
+        dispatch({ payload: user, type: LOGIN_REQUEST });
```
Comment (justiceotuya) [inline src/components/signin/components/AntdForm.jsx]: The aim of sagas is to make side effect calls. Plese move to sagas

Comment (justiceotuya) [reply src/components/signin/components/AntdForm.jsx]: If you are having issue about how to approach sagas, check the code of the past cohort ans see how it is used

File: src/components/signin/actions.js (hunk 4 above)
Comment (edbella) [reply src/components/signin/actions.js]: @justiceotuya But I am calling this particular function in sagas. It is just defined here.

File: src/components/signin/sagas.js [hunk 6]
```diff
@@ -23,11 +23,12 @@ const {
...
  */
-function* loginUser(email, password) {
+function* loginUser({ payload }) {
     try {
-        const user = yield call(loginToFirebase.auth().signInWithEmailAndPassword(email, password));
-        const userDetails = yield put(receiveLogin(user));
-        yield put(LOGIN_SUCCESS, userDetails);
+        const { user } = yield (loginToFirebase.auth()
+            .signInWithEmailAndPassword(payload.email, payload.password));
+        const userDetails = yield receiveLogin(user);
```
Comment (edbella) [inline src/components/signin/sagas.js]: This is where I am calling the function that I described in my actions.js @justiceotuya

File: src/components/signin/actions.js (hunk 4 above)
Comment (Way2nnadi) [reply src/components/signin/actions.js]: all async operations should be defined and handled in your sagas file.

File: src/components/signin/components/AntdForm.jsx (hunk 5 above)
Comment (Way2nnadi) [reply src/components/signin/components/AntdForm.jsx]: @justiceotuya I do not see any issue with the dispatch call here. Please explain, your thought process

File: src/components/signin/sagas.js (hunk 6 above)
Comment (Way2nnadi) [reply src/components/signin/sagas.js]: move that function to this file

File: src/App.jsx (hunk 2 above)
Comment (Way2nnadi) [reply src/App.jsx]:
```
// define in you constants.js file
const routes  = [
    { path: '/', component: Homepage },
    { path: '/sign-up', component: Signup },
    { path: '/sign-in', component: Signin },
];

// Use in your App.jsx file
routes.map(route => <Route exact {...route} />);
```

Comment (edbella) [reply src/App.jsx]: Ok.

**Comment by edbella** [comment]
@justiceotuya @enyeInc @olajohn-ajiboye Please review this PR.

**Pull request merged**